server/runtime/src/main/server/bin/user-tool.sh

```diff
@@ -0,0 +1,251 @@
+       case $p in
+         -Dinfinispan.server.base.dir=*)
+              ISPN_BASE_DIR=`cd ${p#*=} ; pwd -P`
+              ;;
+         -Dinfinispan.server.log.dir=*)
+              if [ -d "${p#*=}" ]; then
+                ISPN_LOG_DIR=`cd ${p#*=} ; pwd -P`
+             else
+                #since the specified directory doesn't exist we don't validate it
+                ISPN_LOG_DIR=${p#*=}
+             fi
+             ;;
+         -Dinfinispan.server.config.dir=*)
+              ISPN_CONFIG_DIR=`cd ${p#*=} ; pwd -P`
+              ;;
+       esac
+    done
+fi
+
+# determine the default base dir, if not set
+if [ "x$ISPN_BASE_DIR" = "x" ]; then
+   ISPN_BASE_DIR="$ISPN_HOME/server"
+fi
+# determine the default log dir, if not set
+if [ "x$ISPN_LOG_DIR" = "x" ]; then
+   ISPN_LOG_DIR="$ISPN_BASE_DIR/log"
+fi
+# determine the default configuration dir, if not set
+if [ "x$ISPN_CONFIG_DIR" = "x" ]; then
+   ISPN_CONFIG_DIR="$ISPN_BASE_DIR/conf"
+fi
+
+# For Cygwin, switch paths to Windows format before running java
+if $cygwin; then
+    ISPN_HOME=`cygpath --path --windows "$ISPN_HOME"`
+    JAVA_HOME=`cygpath --path --windows "$JAVA_HOME"`
+    ISPN_MODULEPATH=`cygpath --path --windows "$ISPN_MODULEPATH"`
+    ISPN_BASE_DIR=`cygpath --path --windows "$ISPN_BASE_DIR"`
+    ISPN_LOG_DIR=`cygpath --path --windows "$ISPN_LOG_DIR"`
+    ISPN_CONFIG_DIR=`cygpath --path --windows "$ISPN_CONFIG_DIR"`
+fi
+
+if [ "$PRESERVE_JAVA_OPTS" != "true" ]; then
+    # Check for -d32/-d64 in JAVA_OPTS
+    JVM_D64_OPTION=`echo $JAVA_OPTS | $GREP "\-d64"`
+    JVM_D32_OPTION=`echo $JAVA_OPTS | $GREP "\-d32"`
+
+    # Check If server or client is specified
+    SERVER_SET=`echo $JAVA_OPTS | $GREP "\-server"`
+    CLIENT_SET=`echo $JAVA_OPTS | $GREP "\-client"`
+
+    if [ "x$JVM_D32_OPTION" != "x" ]; then
+        JVM_OPTVERSION="-d32"
+    elif [ "x$JVM_D64_OPTION" != "x" ]; then
+        JVM_OPTVERSION="-d64"
+    elif $darwin && [ "x$SERVER_SET" = "x" ]; then
+        # Use 32-bit on Mac, unless server has been specified or the user opts are incompatible
+        "$JAVA" -d32 $JAVA_OPTS -version > /dev/null 2>&1 && PREPEND_JAVA_OPTS="-d32" && JVM_OPTVERSION="-d32"
+    fi
+
+    if [ "x$CLIENT_SET" = "x" -a "x$SERVER_SET" = "x" ]; then
+        # neither -client nor -server is specified
+        if $darwin && [ "$JVM_OPTVERSION" = "-d32" ]; then
+            # Prefer client for Macs, since they are primarily used for development
+            PREPEND_JAVA_OPTS="$PREPEND_JAVA_OPTS -client"
+        else
+            PREPEND_JAVA_OPTS="$PREPEND_JAVA_OPTS -server"
+        fi
+    fi
+
+    JAVA_OPTS="$PREPEND_JAVA_OPTS $JAVA_OPTS"
+fi
+
+CLASSPATH=
+for JAR in "$ISPN_HOME/boot"/*.jar; do
+    CLASSPATH="$CLASSPATH:$JAR"
+done
+
+eval \"$JAVA\" $JAVA_OPTS \
+   -Djava.util.logging.manager=org.jboss.logmanager.LogManager \
+   -Dinfinispan.server.home.path=\""$ISPN_HOME"\" \
+   -classpath "$CLASSPATH" "$LOADER_CLASS" "$MAIN_CLASS" "$ARGUMENTS"
```

Comment:
There is a lot of overlap between `user-tool.sh` and `server.sh`. I'm happy for it to go into a future Jira, but I think it makes sense for us to have some kind of `common.sh` that exposes the common code as functions.

Reply:
Yes, I didn't want to go down that path yet, but it will need to be done